### PR TITLE
fix sgd bug and add demo notebook

### PR DESCRIPTION
**Pull request by thecharlieblake**

### CODE REVIEW

File: unit_scaling/tests/test_optim.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def test_scaled_parameters(opt: str) -> None:
     base_wd = 0.001
     param_groups = scaled_parameters(
         model.parameters(),
-        dict(sgd=lr_scale_func_sgd, adam=lr_scale_func_adam)[opt],
+        dict(sgd=lr_scale_func_sgd("to_output_scale"), adam=lr_scale_func_adam)[opt],
```
Comment (DouglasOrr) [inline unit_scaling/tests/test_optim.py:51]: I wonder if we should instead change the final layer to a `LinearReadout` and use this to test the new behaviour?

**Review by DouglasOrr — APPROVED** [review]
A minor query about whether we could use an existing test to verify the new behaviour, but otherwise LGTM.

The notebook is excellent & very clear - thank you for that!

File: unit_scaling/tests/test_optim.py (hunk 1 above)
Comment (thecharlieblake) [reply unit_scaling/tests/test_optim.py]: I've done this, and corrected the input scaling rule which I hadn't actually adjusted. It then clicked that in this case it now (of course!) reduces to the adam rules, so I've modified the code to reflect this.

**Review by DouglasOrr — APPROVED** [review]
Brill, thanks, LGTM!

**Pull request merged**